src/Autofac.Pooling/PoolActivator.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Autofac.Core;
+using Autofac.Core.Resolving.Pipeline;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Autofac.Pooling
+{
+    /// <summary>
+    /// An activator for creating new <see cref="ObjectPool{TLimit}"/> instances.
+    /// </summary>
+    /// <typeparam name="TLimit">The limit type of the objects in the pool.</typeparam>
+    internal sealed class PoolActivator<TLimit> : IInstanceActivator
+        where TLimit : class
+    {
+        private readonly Service _pooledInstanceService;
+        private readonly IPooledRegistrationPolicy<TLimit> _policy;
+        private readonly DefaultObjectPoolProvider _poolProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PoolActivator{TLimit}"/> class.
+        /// </summary>
+        /// <param name="pooledInstanceService">The service used to resolve new instances of the pooled registration.</param>
+        /// <param name="policy">The pool policy.</param>
+        public PoolActivator(Service pooledInstanceService, IPooledRegistrationPolicy<TLimit> policy)
+        {
+            _pooledInstanceService = pooledInstanceService;
+            _policy = policy;
+            _poolProvider = new DefaultObjectPoolProvider
+            {
+                MaximumRetained = policy.MaximumRetained
```

Comment:
**This is the magic.** It took me quite a while to figure out that `DefaultObjectPoolProvider` _isn't ours_ and that the importance of the value `MaximumRetained` isn't that it's a maximum number that gets hit and the pool stops providing; it's a maximum number of things we'll keep pooling and the rest are "overflow." [Nothing really to do on this particular line, but I think some docs on `IPooledRegistratoinPolicy<TLimit>.MaximumRetained` to clarify will help.]

Reply:
Yeah, it took a little squinting at the `ObjectPool` implementation to figure it out for me. 

Updated the comments for `MaximumRetained`.